pkg/router/template/plugin.go

```diff
@@ -253,11 +254,41 @@ func peerEndpointsKey(namespacedName ktypes.NamespacedName) string {
 }
 
 // createRouterEndpoints creates openshift router endpoints based on k8s endpoints
-func createRouterEndpoints(endpoints *kapi.Endpoints, excludeUDP bool) []Endpoint {
+func createRouterEndpoints(endpoints *kapi.Endpoints, excludeUDP bool, service *kapi.Service) []Endpoint {
+	// check if this service is currently idled
+	subsets := endpoints.Subsets
+	if _, ok := endpoints.Annotations[unidlingapi.IdledAtAnnotation]; ok && len(endpoints.Subsets) == 0 {
+		if service == nil {
+			// TODO: actually return an error
+			glog.Errorf("no idled service found corresponding to idled endpoints %v/%v", endpoints.Namespace, endpoints.Name)
+			return []Endpoint{}
+		}
+
+		svcSubset := kapi.EndpointSubset{
```

Comment:
This assumes that clusterIP is set, which does not have to be the case.  For instance, Headless services have the field set to "None", so you have to handle and deal with that here, and probably also guard it in "idle" (it's not clear to me that we can safely idle something that is headless with the current code).